docs/upgrade.txt

```diff
@@ -71,6 +71,21 @@ Version 5.x Breaking Changes
 
 This library version introduces the following breaking changes:
 
+- The query builder returns results as objects, instead of as arrays.
+  This change requires that you change array access to property access
+  when interacting with query results.
+
+  The following code shows how to retrieve a query result and access a
+  property from the result object:
+
+  .. code-block:: php
+     :emphasize-lines: 4
+
+     DB::table('accounts')
+         ->where('name', 'Anita Charles')
+         ->first()
+         ->balance
```

Comment:
You could highlight the change with a diff (not sure if diffs are correctly colorized in the docs):

```suggestion
     $document = DB::table('accounts')
         ->where('name', 'Anita Charles')
         ->first();

-    $document['balance'];
+    $document->balance;
```

Preview:
```diff
     $document = DB::table('accounts')
         ->where('name', 'Anita Charles')
         ->first();

-    $document['balance'];
+    $document->balance;
```

Reply:
The only way we can demonstrate this is by using code comments. I can make that change though!